utilities/blob_db/blob_db_impl.cc

```diff
@@ -0,0 +1,2073 @@
+      }
+    }
+
+    blob_files_.insert(std::make_pair(f_iter.first, bfptr));
+  }
+
+  return status;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//
+//
+////////////////////////////////////////////////////////////////////////////////
+Status BlobFile::readFooter_locked(blob_log::BlobLogFooter& bf) {
+
+  if (file_size_ < (blob_log::BlobLogHeader::kHeaderSize +
+    blob_log::BlobLogFooter::kFooterSize))
+    return Status::IOError("File does not have footer", PathName());
+
+  uint64_t footer_offset = file_size_ - blob_log::BlobLogFooter::kFooterSize;
+  // assume that ra_file_reader_ is valid before we enter this
+  assert(ra_file_reader_);
+
+  Slice result;
+  char scratch[blob_log::BlobLogFooter::kFooterSize+10];
+  Status s = ra_file_reader_->Read(footer_offset,
+        blob_log::BlobLogFooter::kFooterSize, &result, scratch);
+
+  if (!s.ok())
+    return s;
+
+  s = bf.DecodeFrom(&result);
+  return s;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Should take a WriteLock on the mutex_
+//
+////////////////////////////////////////////////////////////////////////////////
+void BlobDBImpl::closeRandomAccess_locked(std::shared_ptr<BlobFile>& bfile) {
+  bfile->closeRandomAccess_locked();
+  open_file_count_--;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Write mutex on file needs to be held
+//
+////////////////////////////////////////////////////////////////////////////////
+std::shared_ptr<RandomAccessFileReader> BlobDBImpl::openRandomAccess_locked(
+  std::shared_ptr<BlobFile> &bfile, Env *env, const EnvOptions& env_options) {
+  bool fresh_open = false;
+  auto rar = bfile->openRandomAccess_locked(env, env_options, &fresh_open);
+  if (fresh_open)
+    open_file_count_++;
+  return rar;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//
+//
+////////////////////////////////////////////////////////////////////////////////
+std::shared_ptr<BlobFile> BlobDBImpl::newBlobFile(const std::string& reason)
+{
+  uint64_t file_num = next_file_number_++;
+  auto bfile = std::make_shared<BlobFile>(this, blob_dir_, file_num);
+  Log(InfoLogLevel::INFO_LEVEL, db_options_.info_log,
+    "New blob file created: %s reason='%s'", bfile->PathName().c_str(),
+    reason.c_str());
+  LogFlush(db_options_.info_log);
+  return bfile;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//
+//
+////////////////////////////////////////////////////////////////////////////////
+Status BlobDBImpl::createWriter_locked(std::shared_ptr<BlobFile>& bfile,
+  bool reopen) {
+  std::string fpath(bfile->PathName());
+  std::unique_ptr<WritableFile> wfile;
+
+  Status s = (reopen) ?
+   myenv_->ReopenWritableFile(fpath, &wfile, env_options_) :
+   myenv_->NewWritableFile(fpath, &wfile, env_options_);
+
+  if (!s.ok())
+    return s;
```

Comment:
I reran lint ... it seems for simple increments and returns, format suggests to put in
same line. if (fresh_open) open_file_count_++;

You are right that if you put in next line, you are supposed to enclose with { }